tests/unit/s2n_config_test.c

```diff
@@ -150,16 +151,18 @@ int main(int argc, char **argv)
         }
     };
 
-    /* Test for s2n_config_new() and tls 1.3 behavior */
+    /* Test for s2n_config_new and tls 1.3 behavior */
     {
         if (!s2n_is_in_fips_mode()) {
             struct s2n_config *config = NULL;
             EXPECT_NOT_NULL(config = s2n_config_new());
+            EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, s2n_auto_gen_old_default_security_policy()));
             EXPECT_EQUAL(config->security_policy, default_security_policy);
```

Comment:
> These tests will now break when the default policy is updated. But I guess it makes sense to wait to fix these in another PR to avoid messing with the auto generation?

I think the test is trying to cover "default behavior of configs" (what policy does it get initialized with by default) (see `3`rd point below).
Therefore the auto-gen is actually changing the intent of the test because and should not be applied.

> This file also contains other tests that test the default config itself. Are there similar checks in these kinds of tests that will fail when the policy is swapped? Just want to make sure that they won't start silently testing the old default policy now that the old default policy is being specifically set.

This is a good point but this is how I was thinking about this:
1) If you want to test with the "default" policy you should have specified `set_cipher_preference("default")`
covered in explicit detection

2) if you really want to test with tls_12 and were lazy, you might not have specified anything
implicit detection which this PR covers

3) if you want to test the "default behavior of configs" (what policy does it get initialized with by default)
this is what you are calling out. I would like to think that these type of tests only exist in s2n_config_test and possibly s2n_security_policy_test. I would audit those test files manually. If not then we would need to audit all tests to make sure of this.

Reply:
Yeah I agree with all of this. Though for 1, this PR will still address the case where the test is using the default policy without calling `set_cipher_preferences` right?

For 3, I agree - I'm not aware of other tests that are testing default configs themselves, so it seems reasonable to me to restrict the audit to this file. To make this easier to review, it might help to skip s2n_config_test in the code generation script and manually remove implicit usages in a followup PR.